libs/wire-api/src/Wire/API/MLS/Message.hs

```diff
@@ -361,3 +363,36 @@ mkRemoveProposalMessage priv pub gid epoch ref = maybeCryptoError $ do
             }
   let sig = BA.convert $ sign priv pub (rmRaw tbs)
   pure (Message tbs (MessageExtraFields sig Nothing Nothing))
+
+mkAppAckProposalMessage ::
+  SecretKey ->
+  PublicKey ->
+  GroupId ->
+  Epoch ->
+  KeyPackageRef ->
+  [MessageRange] ->
+  Maybe (Message 'MLSPlainText)
+mkAppAckProposalMessage priv pub gid epoch ref mrs =
+  maybeCryptoError $ mkAppAckProposalMessageFailable priv pub gid epoch ref mrs
+
+mkAppAckProposalMessageFailable ::
```

Comment:
This is only used for testing, couldn't it just be moved to the tests?

Reply:
Maybe this function and `mkRemoveProposalMessage` can be factored to use a common function that signs a `MessageTBS` to `(Message 'MLSPlainText)`.
Both functions use `maybeCryptoError` together with a `pure` which is just [CryptoPassed](https://hoogle.zinfra.io/file/root/.cabal/store/ghc-8.10.7/cryptonite-0.29-28e7ebd201e4a5f09d307c6367c2ccfcc28816639f78d8c6cbc11e17376be535/share/doc/html/src/Crypto.Error.Types.html#CryptoPassed). I think in both functions nothing is acutally checked.